gpiozero/boards.py

```diff
@@ -1345,4 +1358,204 @@ def set_char_layout(self, char, layout):
+        self._display_thread = None
+        
+        super(MultiSevenSegmentDisplay, self).__init__(*led_pins, pwm=pwm, active_high=active_high)
+        if initial_value:
+            self.on()
+    
+    def display(self, message, align_left=True, refresh_delay=0.007):
+        """
+        Display a message on the multi 7 segment display
+
+        :param string message:
+            The message to be displayed. If the 7 segment display has 
+            a decimal points, they will be used in replacement for 
+            full stops in the message.
+            
+        :param bool align_left:
+            If 'True' (the default) the message will be aligned to the left
+            on the display. If 'False' the message will be aligned to the 
+            right.
+            
+        :param float refresh_delay:
+            The time in seconds that each digit will be turned on when 
+            multiplexing the display.
+            The defalut of 0.007 was chosen by trial and error, short enough
+            so there was no flicker, long enough to still be bright, this 
+            value may need to be modified depending on the display.
+        """
+        self._stop_display()
+        message = self._format_message(message, align_left)
+        self._display_thread = GPIOThread(
+            target=self._display, args=(message, align_left, refresh_delay)
+        )
+        self._display_thread.start()
+    
+    def _display(self, message, align_left, refresh_delay):
+        
+        for digit in cycle(range(len(self.digits))):
+            #if we are at the start of the digits, go to the start of the message              
+            if digit == 0:
+                i = 0
+            
+            #display the digit
+            super(MultiSevenSegmentDisplay, self).display(message[i])
+            #is the next digit a full stop?
+            if self.has_decimal_point:
+                if message[i+1:i+2] == ".":
+                    self.decimal_point = True
+                    i += 1
+                else:
+                    self.decimal_point = False
+
+            #turn the digit on and wait
+            self.digits[digit].on()
+            if self._display_thread.stopping.wait(refresh_delay):
+                break
+            self.digits[digit].off()
+
+            i += 1
+            
+    def _format_message(self, message, align_left):
+        #add spaces to decimal points if needed
+        if self.has_decimal_point:
+            output = ""
+            for i in range(len(message)):
+                if message[i] == '.' and (message[i-1] == '.' or i == 0):
+                    output += ' '
+                output += message[i]        
+            message = output
+
+        #validate the messages length
+        if (self.has_decimal_point and len(message) - message.count('.') > len(self.digits)) or (not self.has_decimal_point and len(message) > len(self.digits)): 
```

Comment:
If the display doesnt support a decimal point, it full stop would be considered to be a character in the message. This would generate an error unless the user had created a character layout for "." using `set_char_layout`.